docs/modules/ROOT/pages/explanation/decisions/portal-api.adoc

```diff
@@ -0,0 +1,78 @@
+= {product} Portal API
+
+== Problem
+
+For the xref:explanation/system/details-ui.adoc[{product} Portal] an API endpoint is needed which implements the backend functionality.
+This API endpoint is described in xref:explanation/system/details-external-services.adoc[External Services].
+As documented, this API endpoint connects various third-party backend systems together and presents it in a well-defined uniform way for the Portal UI to be consumed.
+It must be possible to connect various different backend systems which implement the same functionality so that it is a pluggable system and can be re-used by other parties, which might have different backends for the same task.
+{product} API only specifies the interface, the implementation is then user specific.
+
+== Relevant requirements
+
+* xref:references/quality-requirements/maintainability/portal-api-modularity.adoc[Portal API Modularity]
+* xref:references/quality-requirements/reliability/portal-api-data-recoverability.adoc[Portal API Data Recoverability]
+* xref:references/quality-requirements/performance/portal-api-data-freshness.adoc[Portal API Data Freshness]
+
+== Proposals
```

Comment:
The proposals are lacking details for the unknowing user. I'd say we support the proposals with the diagrams (e.g. the one that is currently in External-Services page) and a new one that better displays the interaction between Kubernetes, the user and the 3rd party system.

Reply:
Does it make sense to differentiate the Kubernetes proposal between plain Operator/CRD and one that uses Crossplane under the hood?
Because implementing the proposal with Crossplane adds a technology dependency to Crossplane which is important to mention IMO, worthy of its own decision if we don't mention it as a separate proposal.

Reply:
I'll work out some diagrams later :+1: 

I wouldn't differentiate which framework is used to implement the controller, as it basically all boils down to CRDs, even with Crossplane. The main point of this decision is the use of the Kubernetes API as the contract, the exact implementation of the controller is left to the user of it.

Reply:
> The main point of this decision is the use of the Kubernetes API as the contract, the exact implementation of the controller is left to the user of it.

Agreed, however, it might be worth to list those potential ways of implementation. Mentioning them, would make this less abstract to read.

Reply:
Updated the proposal accordingly.

Reply:
Habemus diagram now :+1: